django_seed/management/commands/seed.py

```diff
@@ -1,40 +1,105 @@
 
-    args = "[appname ...]"
+    def __init__(self, *args, **kwargs):
+        super(Command, self).__init__(*args, **kwargs)
 
-    option_list = AppCommand.option_list + (
-        make_option('--number', dest='number', default=10,
-                    help='number of each model to seed'),
-    )
+        self.include_models = []
+        self.exclude_models = []
 
-    def handle_app_config(self, app_config, **options):
-        if app_config.models_module is None:
-            raise SeederCommandError('You must provide an app to seed')
 
-        try:
-            number = int(options['number'])
-        except ValueError:
-            raise SeederCommandError('The value of --number must be an integer')
+    def add_arguments(self, parser):
+        parser.add_argument('args', metavar='app_label[.ModelName]', nargs='*',
+                            help='Restricts seeding to the specified app_label or app_label.ModelName.')
+        parser.add_argument('-a', '--all', action='store_true', dest='seed_all', default=False,
+                            help="Seed all registered apps.")
+        parser.add_argument('-n', '--number', type=int, default=10, help="Number of each model to seed.")
+        parser.add_argument('-e', '--exclude', dest='exclude', action='append', default=[],
+                            help='An app_label or app_label.ModelName to exclude '
+                                 '(use multiple --exclude to exclude multiple apps/models).')
+
+    def handle(self, *app_labels, **options):
+        number = options.get('number')
+        seed_all = options.get('seed_all')
+        exclude = options.get('exclude')
+
+        # Exclude models to skip
+        self.process_models_from_applist(exclude, self.exclude_model)
+
+        # Add models to seed
+        if len(app_labels) == 0:
+            if seed_all:  # Add all models from all apps
+                for app_config in apps.get_app_configs():
+                    for model in app_config.get_models():
+                        self.add_model(model)
+            else:
+                raise SeederCommandError("You must specify a list of apps or the --all flag")
+        else:
+            self.process_models_from_applist(app_labels, self.add_model)
 
         seeder = Seed.seeder()
 
         # don't diplay warnings about non-timezone aware datetimes
         import warnings
+
         warnings.showwarning = lambda *x: None
 
-        for model in app_config.get_models():
+        for model in self.prioritized_models():
             seeder.add_entity(model, number)
             print('Seeding %i %ss' % (number, model.__name__))
 
         pks = seeder.execute()
         print(pks)
 
+    def add_model(self, model):
+        if model not in self.include_models:
+            self.include_models.append(model)
+
+    def exclude_model(self, model):
+        if model not in self.exclude_models:
+            self.exclude_models.append(model)
```

Comment:
I'm not sure that I like how including and excluding is handled. I think it would be better to include all at first then change the include and exclude function to something like this.

```python
def include_model(self, model):
    if model not in self.models_to_seed:
        self.models_to_seed.append(model)

def exclude_model(self, model):
    if model in self.models_to_seed:
        self.models_to_seed.remove(model)
```

This way there is only one list to maintain and there's no need to the list operation on line 89.